internal/handlers/pg/msg_findandmodify.go

```diff
@@ -271,6 +271,8 @@ func (h *Handler) upsert(ctx context.Context, tx pgx.Tx, docs []*types.Document,
 			}
 		}
 
+		upsert.SetID()
```

Comment:
Hmm, I don't really understand the logic here. The code above has that `if !upsert.Has("_id")` check. Why do we need to set id again?